### DISCUSSION TITLE
Policy and maintenance of ng-clarity

### DISCUSSION THREAD

**Discussion by mauritslourens** [vmware-clarity/ng-clarity#1176] [General]
Hi, I am working for a company in The Netherlands. For one of our products we would like to use ng-clarity, the Angular component framework you guys developed. In order to be able to support our application for the coming 10+ years, we want to be sure that your component framework is still supported and maintained. 

What triggered our concern is the amount of time it takes to get support for a next Angular release. Version 17 is out since 8th of November 2023, and Clarity will probably support it in March 2024, four months later. In the meantime CVE's in Angular 16 will not be fixed in version 16, only in version 17, so we need to be able to upgrade to the latest version. 

What is your policy and vision for ng-clarity? Is it actively used within VMWare and will it therefore be around and maintained for the foreseeable future?

**Comment by williamernest** [comment]
Thanks for reaching out. I understand your concern about basing a project on an open source library without knowing if it will be supported or maintained long into the future. Unfortunately, I don’t think there’s a component library anywhere that will commit to 10 years. Our goal for Clarity has been to build what makes the most sense for our products at VMware, and that hasn’t changed. The library is still actively used and developed and will be for the foreseeable future. If that changes, we’ll notify the community as soon as we can. 

As for the Angular upgrade, if you’re goal is to upgrade to the latest release of Angular as soon as possible after its release, I would recommend a library like Angular Material since they’re part of Angular itself. 

Our goal is to upgrade to support the latest versions of Angular within a reasonable time (we target ~3 months), which we find works well for us and gives us sufficient time to test against the latest Angular version and also make our own set of changes as we improve our library. There may also be times when that timeline is going to be longer than expected. 

Angular 15/16 are still supported for CVE’s according to the Angular [support policy](https://angular.io/guide/releases#actively-supported-versions).
